### PR TITLE
Modernise UI components

### PR DESCRIPTION
**Pull request by timja**
* Replace the legacy bigtable with [jenkins-table](https://weekly.ci.jenkins.io/design-library/Table/)
* Format the date
* Add an empty state

It means the browser window is no longer scrolling horizontal on page loading at 13" screen size	

Before:

image

After:

image

Empty state:

image

### Testing done

Triggered a few CSP violations and cleared them

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/io/jenkins/plugins/csp/ContentSecurityPolicyManagementLink/index.jelly [hunk 1]
```diff
@@ -36,57 +36,71 @@ THE SOFTWARE.
...
+                                    <code><pre>${record.scriptSample}</pre></code>
+                                </td>
+                            </tr>
+                        </j:forEach>
+                    </table>
+
+                </j:when>
+                <j:otherwise>
+                    <!-- TODO: when baseline is at 2.479.x or higher remove the text attribute -->
+                    <l:notice title="${%No reports}" icon="symbol-checkmark-done-outline plugin-ionicons-api" text="${%No reports}" />
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/csp/ContentSecurityPolicyManagementLink/index.jelly:94]: attribute was renamed from text to title in 2.477, I assume as it wasn't used outside of core, so set both for now

**Review by basil — COMMENTED** [review]
Can you please also adapt `CspRule` in `acceptance-test-harness` which is using the legacy `bigtable` class to parse the table. :grimacing:

**Comment by timja** [comment]
> Can you please also adapt `CspRule` in `acceptance-test-harness` which is using the legacy `bigtable` class to parse the table. 😬

Easiest to just change it at release time:

https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/main/java/org/jenkinsci/test/acceptance/plugins/csp/ContentSecurityPolicyReport.java#L17

File: pom.xml [hunk 2]
```diff
@@ -22,6 +22,25 @@
     <name>Content Security Policy Plugin</name>
     <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-${jenkins.baseline}.x</artifactId>
+                <version>3435.v238d66a_043fb_</version>
```
Comment (daniel-beck) [inline pom.xml:30]:
```suggestion
                <version>3696.vb_b_4e2d1a_0542</version>
```

**Comment by daniel-beck** [comment]
@timja Is it deliberate that the `pane` class remains? The header row is a little narrow.

**Comment by timja** [comment]
No not deliberate

**Pull request merged**